pkg/ingester/ingester.go

```diff
@@ -385,6 +393,15 @@ func (i *Ingester) query(ctx context.Context, from, through model.Time, matchers
 			Values: values,
 		})
 		queriedSamples += len(values)
+		queriedSeries++
+
+		if queriedSamples > i.cfg.MaxSamplesPerQuery {
+			return errors.New("exceeded maximum number of samples in a query")
```

Comment:
Pls use httpgrpc.Errorf with a 4xx status code to correctly propagate this to users and not retry.